LDMP/api.py

```diff
@@ -27,22 +27,119 @@
+            iface.messageBar().pushCritical('Trends.Earth', tr_api.tr('Cannot init auth configuration'))
+            return None
+    else:
+        # update existing
+         if not QgsApplication.authManager().updateAuthenticationConfig(currentAuthConfig):
+            iface.messageBar().pushCritical('Trends.Earth', tr_api.tr('Cannot update auth configuration'))
+            return None
+
+    QSettings().setValue("trends_earth/authId", currentAuthConfig.id())
+    return currentAuthConfig.id()
+
+def remove_current_auth_config():
+    authConfigId = QSettings().value("trends_earth/authId", None)
+    if not authConfigId:
+        iface.messageBar().pushCritical('Trends.Earth', tr_api.tr('No authentication set. Do it in Trends.Earth settings'))
+        return None
+    log('remove_current_auth_config with authId {}'.format(authConfigId))
+
+    if not QgsApplication.authManager().removeAuthenticationConfig( authConfigId ):
+        iface.messageBar().pushCritical('Trends.Earth', tr_api.tr('Cannot remove auth configuration with id: {}').format(authConfigId))
+        return False
+
+    QSettings().setValue("trends_earth/authId", None)
+    return True
+
+def get_auth_config(authConfigId=None, warn=True):
+    if not authConfigId:
+        # not set? then retrieve from config if set
+        authConfigId = QSettings().value("trends_earth/authId", None)
+        if not authConfigId:
+            if warn:
+                iface.messageBar().pushCritical('Trends.Earth', tr_api.tr('No authentication set. Do it in Trends.Earth settings'))
+            return None
+    log('get_auth_config with authId {}'.format(authConfigId))
+
+    configs = QgsApplication.authManager().availableAuthMethodConfigs()
+    if not authConfigId in configs.keys():
+        if warn:
+            iface.messageBar().pushCritical('Trends.Earth', tr_api.tr('Cannot retrieve credentials with authId: {} setup correct credentials before').format(authConfigId))
+        return None
+
+    authConfig = QgsAuthMethodConfig()
+    ok = QgsApplication.authManager().loadAuthenticationConfig(authConfigId, authConfig, True)
+    if not ok:
+        if warn:
+            iface.messageBar().pushCritical('Trends.Earth', tr_api.tr('Cannot retrieve credentials with authId: {} setup correct credentials before').format(authConfigId))
+        return None
+    
+    if not authConfig.isValid():
+        if warn:
+            iface.messageBar().pushCritical('Trends.Earth', tr_api.tr('Not valid auth configuration with authId: {}').format(authConfigId))
+        return None
+
+    # check if auth method is the only supported for no
+    if authConfig.method() != 'Basic':
+        if warn:
+            iface.messageBar().pushCritical('Trends.Earth',
+                tr_api.tr('Auth method with authId: {} is {}. Only basic auth is supported by Trend.Earth').format( authConfigId, authConfig.method() ))
+        return None
+    
+    return authConfig
```

Comment:
@luipir I thought this kind of implementation was already available out of the box from the QGIS Authentication system?

Reply:
??? what are you referring about?

Reply:
the whole process of CRUD for auth config ids.

Reply:
In this case Auth system can manage more thatn basic, but the PR try to leave the api.py and server side interacion as before. and that only support basic.
I can just only remove validity here https://github.com/ConservationInternational/trends.earth/pull/362/files#diff-cf2d71fe8784cbc1c711c47d1ff2d1d41357cf9ad2ff7f3f937d55a71ccd6480R122 
Or I misunderstood your question?

Reply:
I was concerned about the defined methods init_auth_config, remove_current_auth_config,  get_user_email and get_auth_config. I thought when dealing the QGIS Auth system, we won't have to define all these functionality and we could achieve the same using the QGIS Auth API for adding, updating and deleting auth_config details including its ids(may be my understanding about the Auth API is not correct).

Reply:
The general idea is to avoid to change actual code flow of the original code... e.g. minimal modifications. so:
1) init_auth_config is the entry point when clicking register... need to create a new (or modify an existing default) auth config, 
the point directly to the new config to facilitate the user (that is done outside apy.py)
2) the same does remove_current_auth_config that is called as before in the same code flow as before when removing a user
3) get_auth_config is a wrapper to avoid to modify current code flow... done just before api.py try to fill requests with PWD
4) get_user_email is as before but using auth manager

e.g. all is using QgsAuthManager but these wrapper try to minimize changes 

I hope is more clear now

Reply:
yes thanks @luipir